x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/Enrich.java

```diff
@@ -29,8 +29,9 @@
 public class Enrich extends UnaryPlan {
     private final Expression policyName;
     private final NamedExpression matchField;
-    private final EnrichPolicyResolution policy;
-    private List<NamedExpression> enrichFields;
+    private final EnrichPolicy policy;
+    private final Map<String, String> concreteIndices; // cluster -> enrich indices
```

Comment:
To avoid having to keep extending the constructor, the existing EnrichPolicyResolution could be refactored to contain the `EnrichPolicy` (which I think was not used because of some dependency/visibility issue) along with the concreteIndices Map and other items that we might add in the future.

Reply:
I understood your suggestion. I tried this before, but the issue is that the enrich physical plan currently doesn't have an EnrichPolicy, and we need to make a dummy for it. Let's revisit this later.